lazy-pages/src/sys/unix.rs

```diff
@@ -109,7 +109,102 @@ where
     }
 }
 
-pub unsafe fn setup_signal_handler<H>() -> io::Result<()>
+use errno::Errno;
+
+#[derive(Debug, Clone, Copy, derive_more::Display)]
+enum ThreadInitError {
+    #[display(fmt = "Cannot get information about old signal stack: {}", _0)]
+    OldStack(Errno),
+    #[display(fmt = "Cannot mmap space for signal stack: {}", _0)]
+    Mmap(Errno),
+    #[display(fmt = "Cannot set new signal stack: {}", _0)]
+    SigAltStack(Errno),
+}
+
+fn init_for_thread_internal() -> Result<(), ThreadInitError> {
+    use core::{mem, ptr};
```

Comment:
I think all statements of function can be moved to the file itself

Reply:
Sorry, can you please explain more?

Reply:
Will done

Reply:
Done